core/libp2p/network/impl/listener_impl.cpp

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/network/impl/listener_impl.hpp"
+
+namespace libp2p::network {
+
+  ListenerImpl::ListenerImpl(
+      std::shared_ptr<peer::AddressRepository> addrrepo,
+      std::shared_ptr<protocol_muxer::ProtocolMuxer> multiselect,
+      std::shared_ptr<network::Router> router,
+      std::shared_ptr<TransportManager> tmgr,
+      std::shared_ptr<ConnectionManager> cmgr)
+      : addrrepo_(std::move(addrrepo)),
+        multiselect_(std::move(multiselect)),
+        router_(std::move(router)),
+        tmgr_(std::move(tmgr)),
+        cmgr_(std::move(cmgr)) {
+    BOOST_ASSERT(addrrepo_ != nullptr);
+    BOOST_ASSERT(multiselect_ != nullptr);
+    BOOST_ASSERT(router_ != nullptr);
+    BOOST_ASSERT(tmgr_ != nullptr);
+    BOOST_ASSERT(cmgr_ != nullptr);
+  }
+
+  bool ListenerImpl::isStarted() const {
```

Comment:
```suggestion
  bool ListenerImpl::isStarted() const noexcept {
```

Reply:
noexcept is useless here, but it makes mocking harder, so wontfix